cake-autorate.sh

```diff
@@ -1688,6 +1699,26 @@ then
 	exit 1
 fi
 
+# validate config entries before loading
+mapfile -t user_config < <(grep -E '^[^(#| )].*=' "${config_path}" | sed -e 's/[\t ]*\#.*//g' -e 's/=.*//g')
+invalid_config=0
+for value in "${user_config[@]}"
+do
+	[[ "${value}" == "config_file_check" ]] && continue
+	# shellcheck disable=SC2076
+	if [[ ! " ${valid_config_entries[*]} " =~ " ${value} " ]]
+	then
+		invalid_config=1
+		log_msg "ERROR" "${value} is an invalid config entry."
```

Comment:
How about: "The value: '${value}' in config file: '${config_path}' is not a valid config entry."